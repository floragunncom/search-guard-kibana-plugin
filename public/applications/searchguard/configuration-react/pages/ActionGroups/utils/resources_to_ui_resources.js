/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { cloneDeep, map, sortBy } from 'lodash';
import { allowedActionsToPermissionsAndActiongroups } from '../../../utils/helpers';

const resourcesToUiResources = (actionGroups) => {
  return sortBy(
    map(cloneDeep(actionGroups), (values, name) => {
      const {
        actiongroups: _actiongroups,
        permissions: _permissions,
      } = allowedActionsToPermissionsAndActiongroups(values.allowed_actions);
      return {
        ...values,
        _id: name,
        _actiongroups,
        _permissions,
      };
    }),
    '_id'
  );
};

export default resourcesToUiResources;
