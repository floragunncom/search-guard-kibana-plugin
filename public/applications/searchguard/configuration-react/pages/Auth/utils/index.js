/* eslint-disable @kbn/eslint/require-license-header */
export { default as resourcesToUiResources } from './resources_to_ui_resources';
export { getSideNavItems } from './get_side_nav_items';
