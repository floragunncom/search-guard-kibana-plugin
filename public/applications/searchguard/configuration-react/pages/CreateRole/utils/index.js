/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export {
  roleToFormik,
  indexPermissionToUiIndexPermission,
  aliasPermissionToUiAliasPermission,
  tenantPermissionToUiTenantPermission,
  indicesToUiIndices,
  actionGroupsToUiClusterIndexTenantActionGroups,
  tenantsToUiTenants,
  excludeIndexPermissionToUiExcludeIndexPermission,
  maskedFieldsToUiMaskedFields,
} from './role_to_formik';
export { formikToRole } from './formik_to_role';
export { useIndexPatterns, indexPatternNames, renderIndexOption } from './use_index_patterns';
export { fieldNamesToUiFieldNames } from './field_names_to_ui_field_names';
export { mappingsToFieldNames } from './mappings_to_field_names';
export { validateMaskedFields } from './validate_masked_fields';
export { validateMaskedFieldsRegexMask } from './validate_masked_fields_regex_mask';
