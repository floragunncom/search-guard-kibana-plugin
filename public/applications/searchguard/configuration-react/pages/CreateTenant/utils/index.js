export { default as tenantToFormik } from './tenant_to_formik';
export { default as formikToTenant } from './formik_to_tenant';
