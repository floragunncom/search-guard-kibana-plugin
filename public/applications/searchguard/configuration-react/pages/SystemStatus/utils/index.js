export { default as getResource } from './get_resource';
