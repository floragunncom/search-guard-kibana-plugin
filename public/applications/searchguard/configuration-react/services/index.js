/* eslint-disable @kbn/eslint/require-license-header */
export { default as LocalStorageService } from './LocalStorageService';
export { default as ElasticsearchService } from './ElasticsearchService';

export { ApiService } from './ApiService';
export { InternalUsersService } from './InternalUsersService';
export { ActionGroupsService } from './ActionGroupsService';
export { TenantsService } from './TenantsService';
export { SgConfigService } from './SgConfigService';
export { RolesService } from './RolesService';
export { RolesMappingService } from './RolesMappingService';
export { ConfigService } from './ConfigService';
