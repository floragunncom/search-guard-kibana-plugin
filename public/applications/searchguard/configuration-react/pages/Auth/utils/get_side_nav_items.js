/* eslint-disable @kbn/eslint/require-license-header */
import { isEmpty, forEach, startCase } from 'lodash';
import { sideNavItem } from '../../../utils/helpers';
import { SIDE_NAV } from './constants';
import { authenticationText, authorizationText } from '../../../utils/i18n/auth';

const AUTHC = 'authc';

export function getSideNavItems({ resources, onSelectSideNavItem, selectedSideNavItemName }) {
  const sideNavItems = [];
  if (isEmpty(resources)) return sideNavItems;

  const authcItems = [];
  const authzItems = [];

  let authcOrder = 0;
  let authzOrder = 0;

  // We must calculate the order if the ES plugin doesn't provide it
  function calcOrder(order, resourceType) {
    if (order) return order;
    if (resourceType === AUTHC) {
      return authcOrder++;
    }
    return authzOrder++;
  }

  forEach(resources, ({ name, order, resourceType }) => {
    const isCategory = name === SIDE_NAV.AUTHORIZATION || name === SIDE_NAV.AUTHENTICATION;
    const isActive = selectedSideNavItemName === name;
    const navItem = sideNavItem({
      id: name,
      text: startCase(name),
      navData: {
        order: calcOrder(order),
      },
      isCategory,
      isActive,
      onClick: () => onSelectSideNavItem(name),
    });

    if (resourceType === AUTHC) {
      authcItems.push(navItem);
    } else {
      authzItems.push(navItem);
    }
  });

  sideNavItems.push(
    sideNavItem({
      id: SIDE_NAV.AUTHENTICATION,
      text: authenticationText,
      navData: {
        items: authcItems.sort((a, b) => a.order - b.order),
      },
      isCategory: true,
    })
  );

  sideNavItems.push(
    sideNavItem({
      id: SIDE_NAV.AUTHORIZATION,
      text: authorizationText,
      navData: {
        items: authzItems.sort((a, b) => a.order - b.order),
      },
      isCategory: true,
    })
  );

  return sideNavItems;
}
