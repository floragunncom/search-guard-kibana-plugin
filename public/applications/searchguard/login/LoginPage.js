/*
 *    Copyright 2021 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React, { useState, useEffect } from 'react';
import dompurify from 'dompurify';
import {
  EuiSpacer,
  EuiFieldText,
  EuiFieldPassword,
  EuiForm,
  EuiFormRow,
  EuiText,
  EuiCallOut,
  EuiImage,
  EuiButton,
  EuiPanel,
  EuiCard,
  EuiIcon,
  EuiFlexGroup,
  EuiFlexItem,
  EuiErrorBoundary,
  EuiLoadingKibana,
} from '@elastic/eui';
import { LicenseWarningCallout } from '../../components';
import { stringCSSToReactStyle } from '../../../utils/cssHelper';

import { API_ROOT } from '../../../utils/constants';

// @todo Move this to the new app
import { sanitizeNextUrlFromFullUrl } from './sanitize_next_url';

export function isInvalid(error) {
  return error !== null;
}

export function UserNameInput({ isInvalid, userName, onChange }) {
  return (
    <EuiFormRow id="sg.username" label="Username" isInvalid={isInvalid}>
      <EuiFieldText
        id="sg.username"
        data-test-subj="sg.username"
        name="userName"
        required={true}
        placeholder="Username"
        value={userName}
        onChange={onChange}
        autoFocus
        icon="user"
        autoComplete="off"
        isInvalid={isInvalid}
      />
    </EuiFormRow>
  );
}

export function UserPasswordInput({ isInvalid, password, onChange }) {
  return (
    <EuiFormRow id="sg.password" label="Password" isInvalid={isInvalid}>
      <EuiFieldPassword
        id="sg.password"
        data-test-subj="sg.password"
        name="password"
        required={true}
        placeholder="Password"
        value={password}
        onChange={onChange}
        autoComplete="off"
        isInvalid={isInvalid}
      />
    </EuiFormRow>
  );
}



export function ErrorCallout({ error, euiFlexItemProps = {} } = {}) {
  if (!error) return null;

  return (
    <EuiFlexItem {...euiFlexItemProps}>
      <EuiErrorBoundary>
        <EuiCallOut
          id="sg.errorMessage"
          data-test-subj="sg.errorMessage"
          title="Error"
          color="danger"
          iconType="alert"
        >
          <EuiText data-test-subj="sg.errorMessage-text">
            <p>{error}</p>
          </EuiText>
        </EuiCallOut>
      </EuiErrorBoundary>
    </EuiFlexItem>
  );
}

export function AuthTypesMenu({ authTypes = [] }) {
  const _authTypes = authTypes.filter((authType) => authType.type !== 'basicauth');
  if (!_authTypes.length) return null;

  return (
    <EuiFlexItem>
      <EuiErrorBoundary>
        <EuiFlexGroup direction="column" gutterSize="m" data-test-sub="sg.login.authMenu">
          {_authTypes.map((auth, index) => {
            return (
              <EuiFlexItem key={index} style={{ minWidth: 400 }}>
                <EuiErrorBoundary>
                  <EuiCard
                    key={index}
                    data-test-subj={`sg.login.authMenu.item.openid`}
                    layout="horizontal"
                    paddingSize="s"
                    icon={<EuiIcon size="xl" type="empty" />}
                    title={auth.title}
                    description=""
                    href={auth.loginURL}
                  />
                </EuiErrorBoundary>
              </EuiFlexItem>
            );
          })}
        </EuiFlexGroup>
      </EuiErrorBoundary>
    </EuiFlexItem>
  );
}

export function HTMLTitle({ text, euiTextProps = {}, HTMLTag = 'h1' } = {}) {
  return (
    <EuiErrorBoundary>
      <EuiText textAlign="center" {...euiTextProps}>
        <HTMLTag
          // eslint-disable-next-line react/no-danger
          dangerouslySetInnerHTML={{ __html: dompurify.sanitize(text) }}
        />
      </EuiText>
    </EuiErrorBoundary>
  );
}

export function BasicLogin({ httpClient, basicLoginConfig, loginPageConfig }) {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [error, setError] = useState(null);
  const [isLoading, setIsLoading] = useState(false);

  if (!basicLoginConfig) return null;

  const loginButtonStyles = stringCSSToReactStyle(loginPageConfig.buttonstyle);

  function redirectToKibana() {
    const nextUrl = sanitizeNextUrlFromFullUrl(
      window.location.href,
      httpClient.http.basePath.basePath
    );
    window.location.href = nextUrl;
  }

  async function handleSubmit() {
    setError(null);
    setIsLoading(true);

    try {
      await httpClient.post(`${API_ROOT}/auth/login`, { username, password });
      redirectToKibana();
    } catch (error) {
      console.error('BasicLogin, handleSubmit', error);

      let _error =
        'An error occurred while checking your credentials, make sure you have an Elasticsearch cluster secured by Search Guard running.';
      if (error.body) error = error.body;

      if (error.statusCode && error.statusCode === 401) {
        _error = 'Invalid username or password, please try again';
      } else if (error.statusCode && error.statusCode === 404) {
        // This happens either when the user doesn't have any valid tenants or roles
        _error = error.message;
      }

      setError(_error);
    }

    setIsLoading(false);
  }

  return (
    <EuiFlexItem>
      <EuiErrorBoundary>
        <EuiPanel grow={false} style={{ maxWidth: 400 }}>
          <EuiFlexGroup direction="column">
            <EuiFlexItem>
              <HTMLTitle
                HTMLTag="p"
                text={basicLoginConfig.message}
                euiTextProps={{ 'data-test-subj': 'sg.login.subTitle' }}
              />
            </EuiFlexItem>
            <EuiFlexItem>
              <form onSubmit={(event) => event.preventDefault()}>
                <EuiForm>
                  <input
                    autoComplete="anyrandomstring"
                    name="hidden"
                    type="text"
                    style={{ display: 'none' }}
                  />
                  <UserNameInput
                    isInvalid={isInvalid(error)}
                    userName={username}
                    onChange={(e) => setUsername(e.target.value)}
                  />
                  <EuiSpacer />

                  <UserPasswordInput
                    isInvalid={isInvalid(error)}
                    password={password}
                    onChange={(e) => setPassword(e.target.value)}
                  />
                  <EuiSpacer size="l" />

                  <EuiButton
                    id="sg.login"
                    data-test-subj="sg.login"
                    fill
                    fullWidth={true}
                    style={loginButtonStyles}
                    onClick={handleSubmit}
                    type="submit"
                    isLoading={isLoading}
                  >
                    Log in
                  </EuiButton>

                </EuiForm>
              </form>
            </EuiFlexItem>
            <ErrorCallout error={error} />
          </EuiFlexGroup>
        </EuiPanel>
      </EuiErrorBoundary>
    </EuiFlexItem>
  );
}

export function BrandImage({ configService, httpClient }) {
  const { showbrandimage: showBrandImage, brandimage: brandImage } = configService.get(
    'searchguard.login'
  );

  if (!showBrandImage) return null;
  return (
    <EuiFlexItem>
      <EuiErrorBoundary>
        <div style={{ maxWidth: '300px' }}>
          <EuiImage
            data-test-subj="sg.login.brandImage"
            alt="Brand image"
            size="fullWidth"
            url={
              brandImage.startsWith('/plugins')
                ? httpClient.http.basePath.basePath + brandImage
                : brandImage
            }
          />
        </div>
      </EuiErrorBoundary>
    </EuiFlexItem>
  );
}

export function authTypesToUiAuthTypes(authTypes, { basePath = '' } = {}) {
  return authTypes.map((authType) => {
    return {
      ...authType, loginURL: extendLoginURL(authType.loginURL)
    };
  });

  function extendLoginURL(loginURL) {
    if (!loginURL) loginURL = basePath + '/login';
    const currURL = new URL(window.location.href);

    try {
      loginURL = new URL(loginURL);
    } catch (error) {
      loginURL = new URL(loginURL, 'http://sgurlplaceholder');
    }

    let nextURL = currURL.searchParams.get('nextUrl');
    if (nextURL) {
      if (currURL.hash) nextURL += currURL.hash;
      loginURL.searchParams.set('nextUrl', nextURL);
    }

    if (loginURL.hostname.startsWith('sgurlplaceholder')) {
      return loginURL.pathname + loginURL.search;
    }

    return loginURL.toString();
  }
}

export function LoginPage({ httpClient, configService }) {
  const [loginPageConfig, setLoginPageConfig] = useState({});
  const [authTypes, setAuthTypes] = useState([]);
  const [basicLoginConfig, setBasicLoginConfig] = useState(null);
  const [isLoading, setIsLoading] = useState(false);
  const [error, setError] = useState(null);

  useEffect(() => {
    fetchData();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);



  async function fetchData() {
    setError(null);
    setIsLoading(true);

    // if session was not terminated by logout, clear any remaining
    // stored paths etc. from previous users, to avoid issues
    // like a non-working default index pattern
    localStorage.clear();
    sessionStorage.clear();

    try {
      const { data } = await httpClient.get(`${API_ROOT}/auth/config`);
      let authTypes = data.authTypes;
      authTypes = authTypesToUiAuthTypes(authTypes, httpClient.http.basePath.basePath);
      console.debug('LoginPage, fetchData, authTypes', authTypes);
      setLoginPageConfig(data.loginPage);
      setAuthTypes(authTypes);
      setBasicLoginConfig(authTypes.find(({ type }) => type === 'basicauth'));
    } catch (error) {
      console.error('LoginPage, fetchData', error);
      setError(error.message);
    }

    setIsLoading(false);
  }

  return (
    <div style={{ padding: 100 }}>
      <EuiFlexGroup direction="column" alignItems="center" justifyContent="center">
        <BrandImage configService={configService} httpClient={httpClient} />
        <EuiFlexItem>
          <HTMLTitle
            HTMLTag="h2"
            text={loginPageConfig.title}
            euiTextProps={{ 'data-test-subj': 'sg.login.title' }}
          />
        </EuiFlexItem>
        <ErrorCallout error={error} euiFlexItemProps={{ style: { minWidth: 400 } }} />
        <LicenseWarningCallout
          configService={configService}
          euiFlexItemProps={{ style: { minWidth: 400 } }}
        />
        <EuiFlexItem>
          {isLoading ? (
            <EuiLoadingKibana size="xl" />
          ) : (
            <EuiFlexGroup gutterSize="m">
              <BasicLogin
                basicLoginConfig={basicLoginConfig}
                configService={configService}
                httpClient={httpClient}
                loginPageConfig={loginPageConfig}
              />
              <AuthTypesMenu authTypes={authTypes} />
            </EuiFlexGroup>
          )}
        </EuiFlexItem>
      </EuiFlexGroup>
    </div>
  );
}
