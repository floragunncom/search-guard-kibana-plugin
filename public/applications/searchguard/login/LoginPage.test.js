/** @jest-environment jsdom */
/*
 *    Copyright 2020 floragunn GmbH
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { LoginPage } from './LoginPage';
import { ConfigService } from '../../../services';
import { setupApiServiceMock, setupCoreMock, setupCoreContextMock } from '../../../utils/mocks';

describe(LoginPage.name, () => {
  let httpClient;
  let renderer;
  let basePath;
  let systeminfo;
  let uiSettings;
  let coreContext;
  let apiService;

  beforeEach(() => {
    uiSettings = setupCoreMock({
      uiSettingsGet: (setting) => setting !== 'theme:darkMode',
    }).uiSettings;
    apiService = setupApiServiceMock();
    httpClient = () => null;
    renderer = new ShallowRenderer();
    basePath = 'abc';
    systeminfo = {
      sg_license: {
        type: 'TRIAL',
        issue_date: '2021-01-13',
        expiry_date: '2021-03-14',
      },
    };
  });

  test('can render the page with a license and the basicauth login defaults', () => {
    const configGet = jest.fn(() => ({
      basicauth: {
        alternative_login: {
          headers: [],
          show_for_parameter: '',
          valid_redirects: [],
          button_text: 'Login with provider',
          buttonstyle: '',
        },
        login: {
          title: 'Please login to Kibana',
          subtitle:
            'If you have forgotten your username or password, please ask your system administrator',
          showbrandimage: true,
          brandimage: 'plugins/searchguard/assets/searchguard_logo.svg',
          buttonstyle: '',
        },
      },
    }));
    coreContext = setupCoreContextMock({ configGet });

    const configService = new ConfigService({
      uiSettings,
      coreContext,
      apiService,
      config: {
        systeminfo,
      },
    });

    const tree = renderer.render(
      <LoginPage httpClient={httpClient} basePath={basePath} configService={configService} />
    );
    expect(tree).toMatchSnapshot();
  });

  test('can render the page with HTML in login title and subtitle', () => {
    const configGet = jest.fn(() => ({
      basicauth: {
        alternative_login: {
          headers: [],
          show_for_parameter: '',
          valid_redirects: [],
          button_text: 'Login with provider',
          buttonstyle: '',
        },
        login: {
          title: '<h1 style="color:blue;text-align:center;">Login</h1>',
          subtitle:
            '<div style="color:red;background:#ffe5e5;"><ul><li>abc</li><li>def</li></ul></div>',
          showbrandimage: true,
          brandimage: 'plugins/searchguard/assets/my_logo.png',
          buttonstyle: 'background-color:#4CAF50;border:none;color:white;font-size:16px;',
        },
      },
    }));
    coreContext = setupCoreContextMock({ configGet });

    const configService = new ConfigService({
      uiSettings,
      coreContext,
      apiService,
      config: {
        systeminfo,
      },
    });

    const tree = renderer.render(
      <LoginPage httpClient={httpClient} basePath={basePath} configService={configService} />
    );
    expect(tree).toMatchSnapshot();
  });
});
