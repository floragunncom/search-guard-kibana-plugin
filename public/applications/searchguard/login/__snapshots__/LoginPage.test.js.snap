// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage can render the page with HTML in login title and subtitle 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <EuiPanel
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "auto",
          "maxWidth": "300px",
        }
      }
    >
      <EuiImage
        alt="Brand image"
        data-test-subj="sg.login.brandImage"
        size="fullWidth"
        url="plugins/searchguard/assets/my_logo.png"
      />
    </div>
    <EuiText
      data-test-subj="sg.login.title"
      textAlign="center"
    >
      <h2
        dangerouslySetInnerHTML={
          Object {
            "__html": "<h1 style=\\"color:blue;text-align:center;\\">Login</h1>",
          }
        }
      />
    </EuiText>
    <EuiText
      data-test-subj="sg.login.subTitle"
      textAlign="center"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "<div style=\\"color:red;background:#ffe5e5;\\"><ul><li>abc</li><li>def</li></ul></div>",
          }
        }
      />
    </EuiText>
    <LicenseWarningCallout
      configService={
        UiConfigService {
          "apiService": undefined,
          "config": Object {
            "authinfo": Object {},
            "elasticsearch": Object {},
            "kibana": Object {},
            "restapiinfo": Object {},
            "searchguard": Object {
              "basicauth": Object {
                "alternative_login": Object {
                  "button_text": "Login with provider",
                  "buttonstyle": "",
                  "headers": Array [],
                  "show_for_parameter": "",
                  "valid_redirects": Array [],
                },
                "login": Object {
                  "brandimage": "plugins/searchguard/assets/my_logo.png",
                  "buttonstyle": "background-color:#4CAF50;border:none;color:white;font-size:16px;",
                  "showbrandimage": true,
                  "subtitle": "<div style=\\"color:red;background:#ffe5e5;\\"><ul><li>abc</li><li>def</li></ul></div>",
                  "title": "<h1 style=\\"color:blue;text-align:center;\\">Login</h1>",
                },
              },
            },
            "server": Object {},
            "systeminfo": Object {
              "sg_license": Object {
                "expiry_date": "2021-03-14",
                "issue_date": "2021-01-13",
                "type": "TRIAL",
              },
            },
          },
          "coreContext": undefined,
          "uiSettings": undefined,
        }
      }
      errorMessage=""
    />
    <form
      onSubmit={[Function]}
    >
      <EuiForm>
        <input
          autoComplete="anyrandomstring"
          name="hidden"
          style={
            Object {
              "display": "none",
            }
          }
          type="text"
        />
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          id="sg.username"
          isInvalid={false}
          label="Username"
          labelType="label"
        >
          <EuiFieldText
            autoComplete="off"
            autoFocus={true}
            data-test-subj="sg.username"
            icon="user"
            id="sg.username"
            isInvalid={false}
            name="userName"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            value=""
          />
        </EuiFormRow>
        <EuiSpacer />
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          id="sg.password"
          isInvalid={false}
          label="Password"
          labelType="label"
        >
          <EuiFieldPassword
            autoComplete="off"
            compressed={false}
            data-test-subj="sg.password"
            fullWidth={false}
            id="sg.password"
            isInvalid={false}
            isLoading={false}
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            value=""
          />
        </EuiFormRow>
        <EuiSpacer
          size="l"
        />
        <EuiButton
          data-test-subj="sg.login"
          fill={true}
          fullWidth={true}
          id="sg.login"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "#4CAF50",
              "border": "none",
              "color": "white",
              "fontSize": "16px",
            }
          }
          type="submit"
        >
          Log in
        </EuiButton>
      </EuiForm>
    </form>
  </EuiPanel>
</div>
`;

exports[`LoginPage can render the page with a license and the basicauth login defaults 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": "100vh",
      "justifyContent": "center",
    }
  }
>
  <EuiPanel
    style={
      Object {
        "maxWidth": "350px",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "auto",
          "maxWidth": "300px",
        }
      }
    >
      <EuiImage
        alt="Brand image"
        data-test-subj="sg.login.brandImage"
        size="fullWidth"
        url="plugins/searchguard/assets/searchguard_logo.svg"
      />
    </div>
    <EuiText
      data-test-subj="sg.login.title"
      textAlign="center"
    >
      <h2
        dangerouslySetInnerHTML={
          Object {
            "__html": "Please login to Kibana",
          }
        }
      />
    </EuiText>
    <EuiText
      data-test-subj="sg.login.subTitle"
      textAlign="center"
    >
      <p
        dangerouslySetInnerHTML={
          Object {
            "__html": "If you have forgotten your username or password, please ask your system administrator",
          }
        }
      />
    </EuiText>
    <LicenseWarningCallout
      configService={
        UiConfigService {
          "apiService": undefined,
          "config": Object {
            "authinfo": Object {},
            "elasticsearch": Object {},
            "kibana": Object {},
            "restapiinfo": Object {},
            "searchguard": Object {
              "basicauth": Object {
                "alternative_login": Object {
                  "button_text": "Login with provider",
                  "buttonstyle": "",
                  "headers": Array [],
                  "show_for_parameter": "",
                  "valid_redirects": Array [],
                },
                "login": Object {
                  "brandimage": "plugins/searchguard/assets/searchguard_logo.svg",
                  "buttonstyle": "",
                  "showbrandimage": true,
                  "subtitle": "If you have forgotten your username or password, please ask your system administrator",
                  "title": "Please login to Kibana",
                },
              },
            },
            "server": Object {},
            "systeminfo": Object {
              "sg_license": Object {
                "expiry_date": "2021-03-14",
                "issue_date": "2021-01-13",
                "type": "TRIAL",
              },
            },
          },
          "coreContext": undefined,
          "uiSettings": undefined,
        }
      }
      errorMessage=""
    />
    <form
      onSubmit={[Function]}
    >
      <EuiForm>
        <input
          autoComplete="anyrandomstring"
          name="hidden"
          style={
            Object {
              "display": "none",
            }
          }
          type="text"
        />
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          id="sg.username"
          isInvalid={false}
          label="Username"
          labelType="label"
        >
          <EuiFieldText
            autoComplete="off"
            autoFocus={true}
            data-test-subj="sg.username"
            icon="user"
            id="sg.username"
            isInvalid={false}
            name="userName"
            onChange={[Function]}
            placeholder="Username"
            required={true}
            value=""
          />
        </EuiFormRow>
        <EuiSpacer />
        <EuiFormRow
          describedByIds={Array []}
          display="row"
          fullWidth={false}
          hasChildLabel={true}
          hasEmptyLabelSpace={false}
          id="sg.password"
          isInvalid={false}
          label="Password"
          labelType="label"
        >
          <EuiFieldPassword
            autoComplete="off"
            compressed={false}
            data-test-subj="sg.password"
            fullWidth={false}
            id="sg.password"
            isInvalid={false}
            isLoading={false}
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            value=""
          />
        </EuiFormRow>
        <EuiSpacer
          size="l"
        />
        <EuiButton
          data-test-subj="sg.login"
          fill={true}
          fullWidth={true}
          id="sg.login"
          onClick={[Function]}
          style={Object {}}
          type="submit"
        >
          Log in
        </EuiButton>
      </EuiForm>
    </form>
  </EuiPanel>
</div>
`;
